### ISSUE TITLE
New rule S116: Field names should comply with a naming convention

### ISSUE THREAD

**Issue by cristian-ambrosini-sonarsource** [SonarSource/sonar-dotnet#7615]
Implement [S116](https://sonarsource.github.io/rspec/#/rspec/S116/csharp) for C# and VBNET.

**Why is this an issue?**
Sharing some naming conventions is a key point to make it possible for a team to efficiently collaborate. This rule allows to check that field names match a provided regular expression.

**_Noncompliant code example_**
With the default regular expression ^[a-z][a-zA-Z0-9]*$ :
```cs
class MyClass {
   private int my_field;
}
```
**_Compliant solution_**
```cs
class MyClass {
   private int myField;
}
```

**Comment by pavel-mikula-sonarsource** [comment]
Internal ticket [NET-1714](https://sonarsource.atlassian.net/browse/NET-1714)

[NET-1714]: https://sonarsource.atlassian.net/browse/NET-1714?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

**Comment by pavel-mikula-sonarsource** [comment]
Moved
